KMeans.java

```diff
@@ -0,0 +1,126 @@
+import java.io.*;
+import java.util.*;
+import java.lang.Math; 
+
+public class KMeans {
+
+	private Vector midpoint (List<Vector> listofvectors) { //computes + returns a midpoint Vector from a given list of Vectors
+		
+		List<Double> mp_coordinates = new ArrayList<Double>();
+		double number_of_vectors = listofvectors.size();
+		double length_of_vector = listofvectors.get(0).numbers.size();	
+			
+		for (int dimension = 0; dimension < length_of_vector; dimension ++) {
+			
+			double coordinate = 0;
+
+			for (int curr = 0; curr < number_of_vectors; curr ++) {
+				
+				coordinate += listofvectors.get(curr).numbers.get(dimension);
+			}
+				
+			coordinate /= number_of_vectors;
+			mp_coordinates.add(coordinate);
+				
+		}
+		return new Vector(mp_coordinates);
+		
+	}
+	
+	private int closestCenter (Vector start_vector, List<Vector> listofcenters) { // given a starting vector and list of cluster centers, returns closest center's index
```

Comment:
character limit.  And these inline comments need to be replaced with proper Javadoc strings